src/utils/channel_utils.test.js

```diff
@@ -260,4 +261,62 @@ describe('ChannelUtils', () => {
         assert.equal(sortChannelsByDisplayName('en', channelA, channelB), -1);
         assert.equal(sortChannelsByDisplayName('en', channelB, channelA), 1);
     });
+
+    it('sortChannelsByTypeListAndDisplayName', () => {
+        const channelOpen1 = {
+            name: 'channelA',
+            team_id: 'teamId',
+            display_name: 'Unit Test channelA',
+            type: General.OPEN_CHANNEL,
+            delete_at: 0,
+            total_msg_count: 0,
+        };
+
+        const channelOpen2 = {
+            name: 'channelB',
+            team_id: 'teamId',
+            display_name: 'Unit Test channelB',
+            type: General.OPEN_CHANNEL,
+            delete_at: 0,
+            total_msg_count: 0,
+        };
+
+        const channelPrivate = {
+            name: 'channelC',
+            team_id: 'teamId',
+            display_name: 'Unit Test channelC',
+            type: General.PRIVATE_CHANNEL,
+            delete_at: 0,
+            total_msg_count: 0,
+        };
+
+        const channelDM = {
+            name: 'channelD',
+            team_id: 'teamId',
+            display_name: 'Unit Test channelD',
+            type: General.DM_CHANNEL,
+            delete_at: 0,
+            total_msg_count: 0,
+        };
+
+        const channelGM = {
+            name: 'channelE',
+            team_id: 'teamId',
+            display_name: 'Unit Test channelE',
+            type: General.GM_CHANNEL,
+            delete_at: 0,
+            total_msg_count: 0,
+        };
+
+        let sortfn = sortChannelsByTypeListAndDisplayName.bind(null, 'en', [General.OPEN_CHANNEL, General.PRIVATE_CHANNEL, General.DM_CHANNEL, General.GM_CHANNEL]);
+        let actual = JSON.stringify([channelOpen1, channelPrivate, channelDM, channelGM, channelOpen2].sort(sortfn));
```

Comment:
Why stringify? I think you can compare arrays in jest without convert them to string. 0/5

Reply:
jest complains that `Compared values have no visual difference.` 

According to https://github.com/facebook/jest/issues/5998 some object literals may not work correct with `equal`.

Reply:
According to the link that happens with annonymous funtions, no?

Reply:
Instead of `assert.equal`, what if you use `expect(actual).toEqual(expected)`? I don't remember if `assert.equal` expects both objects to be the same object or if it checks that they're deeply equal. With `expect`, there's `toEqual` for deep equality or `toBe` for ensuring that they're the same object

Reply:
`expect(actual).toEqual(expected)`  works. So does `assert.deepEqual`. I was incorrectly using `assert.Equal`.